avm-transpiler/src/transpile.rs

```diff
@@ -1116,6 +1120,39 @@ fn handle_storage_read(
     })
 }
 
+/// Patch a Noir function's debug info with updated PCs since transpilation injects extra
+/// instructions in some cases.
+pub fn patch_debug_info_pcs(
+    debug_infos: &Vec<DebugInfo>,
+    brillig_pcs_to_avm_pcs: &Vec<usize>,
+) -> Vec<DebugInfo> {
+    let mut patched_debug_infos = debug_infos.clone();
+
+    for patched_debug_info in patched_debug_infos.iter_mut() {
+        // create a new map with all of its keys (OpcodeLocations) patched
+        let mut patched_locations: BTreeMap<OpcodeLocation, Vec<Location>> = BTreeMap::new();
```

Comment:
Do you just need a map? Why not HashMap? (I think it exists? I think in general tree-based maps are only used when order matters, and might have worse time complexity). But I'll defer to any Rust expert!

(ahh... is it because it's what DebugInfo uses?)

Reply:
Yep I'm just overwriting the entries already in DebugInfo! Not sure why it's a BTreeMap